examples/neighbors/plot_nca_dim_reduction.py

```diff
@@ -81,8 +82,12 @@
     plt.figure()
     # plt.subplot(1, 3, i + 1, aspect=1)
 
-    # Fit the method's model
-    model.fit(X_train, y_train)
+    # There are some hard to avoid collinearity warnings that get raised
+    # by the LinearDiscriminantAnalysis
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
```

Comment:
I would only catch the specific warning, and not all of them, otherwise lgtm.

Reply:
@adrinjalali why not just remove the warning from the code? #14361 ?

Reply:
I'd be more than happy with removing the warning!